internal/example/main/main.go

```diff
@@ -70,6 +70,6 @@ func main() {
 	// Run the xDS server
 	ctx := context.Background()
 	cb := &testv3.Callbacks{Debug: l.Debug}
-	srv := serverv3.NewServer(ctx, cache, cb)
+	srv := serverv3.NewServer(ctx, cache, cb, l)
```

Comment:
Understood, I did it because I needed to add some logging within the server delta handlers but if that needs to be removed I can take it out.

Reply:
Yeah, my point isn't against logging, my point was that this is a signature change to a "public" method (i.e., one in the `pkg` not `internal` directory) in a library.  As such, this is a breaking change.  Hopefully the rest of this functionality can be considered an added capability, but this one is clearly a change and will break downstream builds (likely all of them).

I realize this repository is not yet at 1.0.0 and this may not be a concern yet but the logger injection is not, strictly speaking, a requirement for incremental xDS.